### PR TITLE
Add BitDropDown This close #258

### LINKED ISSUE #258: Add DropDown Component
Add **DropDown** Component based on [Microsoft FluentUI Design](https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown).

### PR DESCRIPTION
**Pull request by mirza-developer**
this closes #258

### CODE REVIEW

**Review by msynk — CHANGES_REQUESTED** [review]

**Comment by msynk** [comment]
please resolve the conflicts too :)

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor [hunk 1]
```diff
@@ -317,25 +317,6 @@
     <BitButton OnClick="() => RatingOutsideValue = 0">Make Rating Empty</BitButton>
 </div>
 
-<hr />
```
[the remark this answers is not in the document]
Comment (mirza-developer) [reply src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor:323]: I can't understand this comment
Please tell more detail

[the remark this answers is not in the document]
Comment (msynk) [reply src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Component.razor:323]: The changes shown here are related to another component, if it is not actually changed and the diff tools showing it wrongly, ignore this comment.

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitDropDownTest.razor [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        IncrementCount();
+    }
+    public int CurrentCount { get; set; }
+    public void IncrementCount()
+    {
+        CurrentCount++;
+        Value = CurrentCount.ToString();
+    }
+    }
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitDropDownTest.razor:31]: code formatting issue

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitInputTests.cs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿using System.Linq;
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitInputTests.cs:1]: This file should not be changed in this PR, please move the required changes into another branch and submit a new PR for a related issue.

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitDropDownTests.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+using Bunit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bit.Client.Web.BlazorUI.Tests.Inputs
+{
+    [TestClass]
+    public class BitDropDownTests : BunitTestContext
+    {
+        [DataTestMethod,
+         DataRow(Visual.Fluent,true,1,true,"enabled"),
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitDropDownTests.cs:13]: code formatting issues

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitDropDown.razor.cs [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            {
+                expandClass = value;
+                ClassBuilder.Reset();
+            }
+        }
+
+        protected override string RootElementClass => "bit-drp";
+        protected override void RegisterComponentClasses()
+        {
+            ClassBuilder.Register(() => string.IsNullOrWhiteSpace(FocusClass)
```
[the remark this answers is not in the document]
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Inputs/BitDropDown.razor.cs:67]: did you miss this one?

**Review by msynk — CHANGES_REQUESTED** [review]
Please resolve the Nullable Reference Type warnings only for the files changed in this PR

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitInputTests.cs (hunk 3 above)
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Inputs/BitInputTests.cs:1]: did you miss rolling back these changes?

**Review by msynk — CHANGES_REQUESTED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**